### DISCUSSION TITLE
Empty `page.git` object in CI... is shallow clone the only possible cause?

### DISCUSSION THREAD

**Discussion by chris-dura** [vuepress/core#1042] [Q&A] [answered]
When I build locally or in dev mode, I'm able to get access to the `page.git` object (eg, for Contributors and Last Updated).
However, when I build the project in CI/CD on our self-hosted GitLab the `page.git` object is empty.

The docs for `@vuepress/plugin-git` mention that typical CI workflows create shallow clones, so it _may_ be that the GitLab CI Runner is creating a shallow clone: https://v2.vuepress.vuejs.org/reference/plugin/git.html#usage

Unfortunately, there is an open issue in GitLab that indicates the settings meant to disable shallow clones are no longer working as they should: https://gitlab.com/gitlab-org/gitlab/-/issues/350100

So, I'm waiting for the DevOps team to verify or confirm a workaround that the Runners are creating unshallow clones.

But, I have a feeling that may not be the issue, and my question is this:

Could the empty `git:{}` be caused by something _other than_ a shallow clone? Is it feasibly possible that the git user the Runner environment is utilizing ONLY has permission to `clone`, and not to access the commit history, or `updatedTime`,  etc?

**Comment by meteorlxy** [comment, marked as answer]
Well, that depends on your runner env. You can try to execute those commands used by the git plugin to check if they could get correctly stdout:

https://github.com/vuepress/vuepress-next/blob/b3565cba73d0e87540b563d86792bbd2bd219de0/ecosystem/plugin-git/src/node/utils/checkGitRepo.ts#L8

https://github.com/vuepress/vuepress-next/blob/b3565cba73d0e87540b563d86792bbd2bd219de0/ecosystem/plugin-git/src/node/utils/getContributors.ts#L8-L15

https://github.com/vuepress/vuepress-next/blob/b3565cba73d0e87540b563d86792bbd2bd219de0/ecosystem/plugin-git/src/node/utils/getUpdatedTime.ts#L10-L23

**Reply by chris-dura** [reply]
fwiw, the root cause was actually that SOMEONE switched from using the `node:16` Docker image to using the `node:16-alpine` Docker image... which is slimmer than slim, and (I believe) does not include `git`... which would probably not be an issue for most of our projects, as running the `npm run build` command doesn't usually require git... but since we're using the `@vuepress/plugin-git`... we need `git` to be installed in the Docker image.

But, @meteorlxy your suggestion helped put me into the context where I could even notice that 😵‍💫 , so thanks!
